### ISSUE TITLE
Long first name can cause phone number and email address to not show on printed directory

### ISSUE THREAD

**Issue by yuy0311** [churchio/onebody#572]
Hi,

I have tested this. Email address if it's from rocketmail.com, it never prints on the directory. All other email addresses are fine. 
Could you investigate it?

Thanks
Yang

**Comment by seven1m** [comment]
Are you talking about the printed directory PDF? If so, do you mean that the family is not printed, or that just the email address isn't printed?

**Comment by yuy0311** [comment]
Hi Tim, I should clarify the issue a bit. It's just the email address is not printed in the directory pdf file. However, if I use other email address, it works fine. As far as I know, the rocketmail.com doesn't work.

**Comment by seven1m** [comment]
Is it possible that the user with the rocketmail address has their email hidden on the Privacy tab of their profile? That might explain why the email address doesn't show.

**Comment by yuy0311** [comment]
@seven1m 

I double checked the privacy setting. Visible attribute is ticked under email column. 

This is how I test for this member. 
Firstly, modify her email address to use rocketmail.com, then use the print directory function. Only her email address is not shown on the pdf. Her name, telephone was displayed correctly.
Second test is then modify her email address to use other email address, I have tested to use gmail.com, yahoo.com, ymail.com, our church domain, random text combination with .com at the end. They are all printed correctly. It's the same privacy setting I use before.  So I am not sure why rocketmail.com was not accepted.

**Comment by seven1m** [comment]
Interesting! Thanks for the extra info... I'll see what I can figure out.

**Comment by seven1m** [comment]
I figured out what is causing this (kind of) -- it seems that a long first name will cause the email address to go out of the printable area. I don't know how to fix this with the pdf-writer gem, which is a very old, abandoned gem.

Anyone is free to try to fix this (the offending code is somewhere [here](https://github.com/churchio/onebody/blob/407f1a58c3cbdc605520c9cfc1f4d8aaddc906f9/app/models/concerns/person/pdf_gen.rb#L99-L106)), but I'd rather focus my attention on rewriting the PDF generation code. I created a separate issue to track that work: #573
